File: rust/gui-client/src-tauri/src/client/about.rs
//! Everything related to the About window

#[tauri::command]
pub(crate) fn get_cargo_version() -> String {
    env!("CARGO_PKG_VERSION").to_string()
}

#[tauri::command]
pub(crate) fn get_git_version() -> String {
    option_env!("GITHUB_SHA").unwrap_or("unknown").to_owned()
}

#[cfg(test)]
mod tests {
    #[test]
    fn version() {
        let cargo = super::get_cargo_version();
        let git = super::get_git_version();

        assert!(cargo != "Unknown", "{}", cargo);
        assert!(cargo.starts_with("1."));
        assert!(cargo.len() >= 2, "{}", cargo);
    }
}

File: rust/tauri-client/src/about.js
import "./tauri_stub.js";
const invoke = window.__TAURI__.tauri.invoke;
const cargoVersionSpan = (document.getElementById("cargo-version"));
const gitVersionSpan = document.getElementById("git-version");
function get_cargo_version() {
    invoke("get_cargo_version")
        .then((cargoVersion) => {
        cargoVersionSpan.innerText = cargoVersion;
    })
        .catch((e) => {
        cargoVersionSpan.innerText = "Unknown";
        console.error(e);
    });
}
function get_git_version() {
    invoke("get_git_version")
        .then((gitVersion) => {
        gitVersionSpan.innerText = gitVersion;
    })
        .catch((e) => {
        gitVersionSpan.innerText = "Unknown";
        console.error(e);
    });
}
document.addEventListener("DOMContentLoaded", () => {
    get_cargo_version();
    get_git_version();
});
